components/buttons.tsx

```diff
@@ -0,0 +1,42 @@
+import type { FC } from "react";
+import React, { type ReactElement } from "react";
+import { signInUserWithGoogle, signOutUser } from "@/firebase/authentication";
+import Button from "@mui/material/Button";
+
+const handleAsyncClick = (asyncFunction: () => Promise<void>) => (): void => {
+  asyncFunction().catch((error: unknown) => {
+    console.error("Error:", error);
```

Comment:
No `console`. This code will run on some server at some point, and we don't need to polute the provider's STDOUT with our messages. Loggers are used for this purpose

Reply:
Has been removed